### PR TITLE
[Ready][Security] Hotfix for CSRF and XSS vulnerabilities

### LINKED ISSUE #5183: Find and fix XSS and GET parameter vulnerabilities
This bounty is to find and fix issues related to XSS and use of GET parameters instead of POST, bypassing CSRF

### PR DESCRIPTION
**Pull request by MajorTomSec**
##### Description

This is a hotfix for critical security vulnerabilities I reported to gitcoin's engineering team (including CSRF and XSS). 

##### Refers/Fixes

This PR partially fixes #5183. 

Deeper code review is still on-going on my side, I will thus add new commits to this PR as I patch new security issues. 
Nevertheless, this PR can be merged as of now in order to fix the reported issues.

##### Testing

I request @danlipert review on this one, as he is the one who followed the issues as I reported them.

### CODE REVIEW

**Comment by ririen** [comment]
Any progress? This PR contains fix for critical issue.

File: app/dashboard/tip_views.py [hunk 1]
```diff
@@ -125,18 +125,22 @@ def receive_tip_v3(request, key, txid, network):
     elif not_mined_yet:
         messages.info(request, f'This tx {tip.txid}, is still mining.  Please wait a moment before submitting the receive form.')
     elif request.GET.get('receive_txid') and not tip.receive_txid:
-        params = request.GET
+        params = request.GET # Should be request.POST, and CSRF protection should be enforced
```
Comment (danlipert) [inline app/dashboard/tip_views.py]: Lets definitely change it to POST then

**Comment by danlipert** [comment]
@MajorTomSec Sorry for missing this one - generally if its marked as WIP we don't start reviewing it, so be sure to remove that from the title. Also, the testing section in the PR form is to describe how the author of PR tested their changes - screenshots and/or video are highly desired to show that the changed functionality still works as normal.

**Comment by thelostone-mc** [comment]
@MajorTomSec would you be able to address the comments on the PR ?

**Comment by MajorTomSec** [comment]
> @MajorTomSec would you be able to address the comments on the PR ?

Sorry, I've been busy with life recently. I have made changes locally and I just require a few more tests to make sure it doesn't break anything. I'll try to update the PR by tomorrow.

**Comment by MajorTomSec** [comment]
This should be ready for merging. Please let me know if you have any comment.

File: app/assets/v2/js/pages/kudos_receive.js [hunk 2]
```diff
@@ -99,10 +99,14 @@ $(document).ready(function() {
       if (err) {
         _alert(err.message.split('\n')[0], 'error');
       } else {
-        document.location.href = window.location.href.split('?')[0] +
-        '?receive_txid=' + txid +
-        '&forwarding_address=' + $('#forwarding_address').val() +
-        '&save_addr=' + ($('#save_addr').is(':checked') ? '1' : '0');
+        var url = window.location.href.split('?')[0];
+        var form = $('<form action="' + url + '" method="post">' +
```
Comment (thelostone-mc) [inline app/assets/v2/js/pages/kudos_receive.js]: `const`

File: app/assets/onepager/js/receive.js [hunk 3]
```diff
@@ -91,10 +91,14 @@ $(document).ready(function() {
       if (err) {
         _alert(err.message.split('\n')[0], 'error');
       } else {
-        document.location.href = window.location.href.split('?')[0] +
-        '?receive_txid=' + txid +
-        '&forwarding_address=' + $('#forwarding_address').val() +
-        '&save_addr=' + ($('#save_addr').is(':checked') ? '1' : '0');
+        var url = window.location.href.split('?')[0];
+        var form = $('<form action="' + url + '" method="post">' +
```
Comment (thelostone-mc) [inline app/assets/onepager/js/receive.js]: `const`

File: app/assets/onepager/js/receive.js [hunk 4]
```diff
@@ -91,10 +91,14 @@ $(document).ready(function() {
       if (err) {
         _alert(err.message.split('\n')[0], 'error');
       } else {
-        document.location.href = window.location.href.split('?')[0] +
-        '?receive_txid=' + txid +
-        '&forwarding_address=' + $('#forwarding_address').val() +
-        '&save_addr=' + ($('#save_addr').is(':checked') ? '1' : '0');
+        var url = window.location.href.split('?')[0];
```
Comment (thelostone-mc) [inline app/assets/onepager/js/receive.js]: `const`

File: app/assets/v2/js/pages/kudos_receive.js [hunk 5]
```diff
@@ -99,10 +99,14 @@ $(document).ready(function() {
       if (err) {
         _alert(err.message.split('\n')[0], 'error');
       } else {
-        document.location.href = window.location.href.split('?')[0] +
-        '?receive_txid=' + txid +
-        '&forwarding_address=' + $('#forwarding_address').val() +
-        '&save_addr=' + ($('#save_addr').is(':checked') ? '1' : '0');
+        var url = window.location.href.split('?')[0];
```
Comment (thelostone-mc) [inline app/assets/v2/js/pages/kudos_receive.js]: `const`

**Comment by thelostone-mc** [comment]
@MajorTomSec looks like travis is failing cause of eslint ! mind fixing those and my comments and we can get this in

**Comment by thelostone-mc** [comment]
^ @MajorTomSec would you be able to get the travis green + address the comments before this PR goes stale ?

**Comment by thelostone-mc** [comment]
PR was moved over to https://github.com/gitcoinco/web/pull/5926!
Thanks @MajorTomSec  ^_^

**Pull request closed without merging**